src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -399,21 +405,34 @@ findNeighborRanks( std::vector< vtkBoundingBox > boundingBoxes )
  * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
  * @param[in] comm the MPI communicator
  * @param[in] partitionRefinement number of graph partitioning refinement cycles
+ * @param[in] useGlobalIds if set to one, use global id arrays from the vtk mesh when available. Else, always recompute them.
  */
 vtkSmartPointer< vtkDataSet >
 redistributeMesh( vtkDataSet & loadedMesh,
                   MPI_Comm const comm,
                   VTKMeshGenerator::PartitionMethod const method,
-                  int const partitionRefinement )
+                  int const partitionRefinement,
+                  int const useGlobalIds )
 {
   GEOSX_MARK_FUNCTION;
 
-  // Generate global IDs for vertices and cells
-  vtkNew< vtkGenerateGlobalIds > generator;
-  generator->SetInputDataObject( &loadedMesh );
-  generator->Update();
-  vtkSmartPointer< vtkDataSet > mesh =
-    vtkDataSet::SafeDownCast( generator->GetOutputDataObject( 0 ) );
+  // Generate global IDs for vertices and cells, if needed
+  vtkSmartPointer< vtkDataSet > mesh;
+  if( useGlobalIds != 1
+      || loadedMesh.GetPointData()->GetGlobalIds() == nullptr
+      || loadedMesh.GetCellData()->GetGlobalIds() == nullptr )
+  {
+    vtkNew< vtkGenerateGlobalIds > generator;
+    generator->SetInputDataObject( &loadedMesh );
+    generator->Update();
+    mesh = vtkDataSet::SafeDownCast( generator->GetOutputDataObject( 0 ) );
+  }
+  else
+  {
+    mesh.TakeReference( &loadedMesh );
+    std::cout << "using global Ids... wish me luck" << std::endl;
```

Comment:
```suggestion
    GEOSX_LOG( "Using global Ids defined in array GLOBAL_IDS." );
```
Plus, can you add small checks to be sure that the fields have the proper dimensions, etc...
If you think this is tricky to use (`wish me luck`), we should be careful I think.

Reply:
Whoops, forgot to change it from debug to logging. Thanks. I will add the basic checks